### PR TITLE
Add cat/alias support for DNFOF 

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
This adds DNFOF support for Cat Aliases API. The behavior prior to this change is that cat alias API would fail with error: no permissions for [indices:admin/aliases/get], even with DNFOF is enabled. Now the behavior is that cat alias API will return the aliases with indices that you have access to. I added a DNFOF test show casing this behavior. 
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Bug fix
* Why these changes are required?
Fix: #4413 
* What is the old behavior before changes and new behavior after changes?
See description

### Issues Resolved
Fix: #4413
Is this a backport? If so, please add backport PR # and/or commits #
No

### Testing
Manual testing, added a test case, fixed a unit test
### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java [hunk 1]
```diff
@@ -443,6 +444,23 @@ public void shouldPerformCatIndices_positive() throws IOException {
         }
     }
 
+    @Test
+    public void shouldPerformCatAliases_positive() throws IOException {
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java:449]: wdyt about performing the same request as the admin user and verify that the response contains `horrible_songs`?

Comment (derek-ho) [reply src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java:442]: Added a test here: [d0b0882](https://github.com/opensearch-project/security/pull/4436/commits/d0b0882ec271d67d651f671640faf52bc1600e52)

File: src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java [hunk 2]
```diff
@@ -48,15 +48,7 @@
...
-import static org.hamcrest.Matchers.aMapWithSize;
-import static org.hamcrest.Matchers.allOf;
-import static org.hamcrest.Matchers.arrayContainingInAnyOrder;
-import static org.hamcrest.Matchers.arrayWithSize;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.hasKey;
-import static org.hamcrest.Matchers.not;
-import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.*;
```
Comment (scrawfor99) [inline src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java:51]: I think we don't want to use wildcard imports.

**Review by scrawfor99 — APPROVED** [review]

File: src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java (hunk 2 above)
Comment (derek-ho) [reply src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java:51]: I ran spotless apply on it and it didn't change anything, should I change it back?

Comment (derek-ho) [reply src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java:51]: Should I create a follow up issue that fails the spotless apply if there is a * import?

Comment (derek-ho) [reply src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java]: Reverted it here: [75877b7](https://github.com/opensearch-project/security/pull/4436/commits/75877b70a414fa7a5921042049ac94a4b49d3b83)

**Review by cwperks — APPROVED** [review]

**Review by scrawfor99 — APPROVED** [review]

**Pull request merged**